tests/cpp/test_resize.cpp

```diff
@@ -3733,4 +3730,522 @@ TEST_F(ResizeTest, SliceScheduledLikeProducer) {
   NVF_CHECK(ref.equal(cg_outputs[0]));
 }
 
+TEST_F(ResizeTest, PadScheduledLikeConsumer) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape({100});
+
+  // concrete shapes to avoid dynamic Fusion
+  auto tv0 = makeConcreteTensor(shape);
+  fusion.addInput(tv0);
+
+  auto tv1 = add(tv0, IrBuilder::create<Val>(1));
+  auto tv2 = pad(tv1, {IrBuilder::create<Val>(1), IrBuilder::create<Val>(1)});
+
+  auto tv3 = add(tv2, IrBuilder::create<Val>(1));
+  fusion.addOutput(tv3);
+
+  auto tv1_padded = IterDomain::resize(
+      tv1->getLoopDomain().at(0),
+      IrBuilder::create<Val>(1, DataType::Index),
+      IrBuilder::create<Val>(1, DataType::Index));
+  tv1->setLoopDomain({tv1_padded});
+
+  for (auto tv : {tv1, tv2, tv3}) {
+    tv->split(0, 32);
+    tv->axis(0)->parallelize(ParallelType::BIDx);
+    tv->axis(1)->parallelize(ParallelType::TIDx);
+  }
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  auto t0 = at::randn(shape, options);
+  std::vector<c10::IValue> aten_inputs({t0});
+
+  EnableOptionsGuard enable_options_guard;
+  EnableOptionsGuard::getCurOptions().set(EnableOption::IdModel, {"all"});
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, aten_inputs);
+  auto cg_outputs = fe.runFusion(aten_inputs);
+
+  auto ref = at::pad(t0 + 1, {1, 1}) + 1;
+
+  NVF_CHECK(ref.equal(cg_outputs[0]));
+}
+
+// Slicing the left half and pad it to the original extent
+TEST_F(ResizeTest, SliceThenPadLeftHalf) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape({100});
+
+  // concrete shapes to avoid dynamic Fusion
+  auto tv0 = makeContigConcreteTensor(shape);
+  fusion.addInput(tv0);
+
+  auto tv1 = set(tv0);
+
+  auto tv2 =
+      slice(tv1, {{fusion.zeroVal(), IrBuilder::create<Val>(shape[0] / 2)}});
+
+  auto tv3 = pad(tv2, {fusion.zeroVal(), IrBuilder::create<Val>(shape[0] / 2)});
+
+  fusion.addOutput(tv3);
+
+  tv2->setLoopDomain(tv2->getRootDomain());
```

Comment:
In this case, we can use the logical domain of the input or the logical domain of the output. Both are equivalent, but that equivalence relationship is not detected by IdModel. Adding some analysis to detect that relationship is part of the overall plan, but it isn't strictly required.

In this test, I'm using the input logical domain, which is mapped with the tv1 logical and the tv2 root domains. Since each tensor needs to use its own IDs, 

> tv2->setLoopDomain(tv2->getRootDomain())

is the simplest way to set the loop domain of tv2.

If the padding width of tv3 is different, then it'd be unclear what loop domain should be used, especially for the right-half case.

Reply:
> If the padding width of tv3 is different, then it'd be unclear what loop domain should be used, especially for the right-half case.

Does this mean that we require the final output of the fusion to be of the same shape (for the slice/pad dimension) with the original input with a `slices+pads` chain?

Is this a requirement on patterns we support? This feels a big strange to have, i.e. looking at the `pad` only test above `PadScheduledLikeConsumer`, we don't have such restriction there.
Moving forward the chains of slice/pad(concat) below, how would we parse this restriction in that case, is the restriction translate to that each matching pair of slice/pad needs to cancel each other?

Reply:
If it's a pad only, it should be fine. But if there's a slice and then a pad, there will be:

1. input domain
2. sliced domain of the input domain
3. padded domain of the sliced domain

The requirement is that #1 and #3 correspond to the same domain. If not, it's unclear what domain to generate a loop for it. Technically, it should be a domain that has an extent of `max(#1, #3)`, but we don't have a capability like this. 

If there's only a pad,

1. input domain
2. padded domain of the input domain

Then we can just use #2 as it's larger than #1.